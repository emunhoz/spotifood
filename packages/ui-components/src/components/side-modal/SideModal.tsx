import React from 'react'
import CloseIcon from './img/close-icon.svg'
import * as S from './SideModal.style'

interface ISideModal {
  show: boolean
  close: () => void
  children: React.ReactChild
}

export const SideModal = ({ show, children, close }: ISideModal) => {
  return (
    <S.ModalContent toogleFilter={show}>
      <S.ModalWrapper>
        <S.CloseModal onClick={() => close()}>
          <S.CloseIcon src={CloseIcon} alt='Fechar' width='44' height='44' />
          Fechar
        </S.CloseModal>
        {children}
      </S.ModalWrapper>
    </S.ModalContent>
  )
}
