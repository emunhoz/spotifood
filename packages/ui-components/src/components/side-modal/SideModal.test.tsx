import React from 'react'
import { render } from '../../utils/test-utils'
import { SideModal } from './SideModal'

describe('[Component]: SideModal', () => {
  test('render the component', () => {
    const { asFragment } = render(
      <SideModal show={false} close={() => console.log}>
        <h1>Hello world</h1>
      </SideModal>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
