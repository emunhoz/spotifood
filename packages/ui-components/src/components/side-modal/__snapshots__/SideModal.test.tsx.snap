// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: SideModal render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM bUwgPB"
  >
    <div
      class="sc-hKFxyN cDmoop"
    >
      <div
        class="sc-gtsrHT pFztO"
      >
        <img
          alt="Fechar"
          class="sc-dlnjwi cyBfdP"
          height="44"
          src="test-file-stub"
          width="44"
        />
        Fechar
      </div>
      <h1>
        Hello world
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
